docs/source/java/examples/quickstartguide.rst

```diff
@@ -0,0 +1,277 @@
+
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.vector.IntVector;
+
+    try(BufferAllocator rootAllocator = new RootAllocator();
+        IntVector intVector = new IntVector("fixed-size-primitive-layout", rootAllocator)){
+        intVector.allocateNew(3);
+        intVector.set(0,1);
+        intVector.setNull(1);
+        intVector.set(2,2);
+        intVector.setValueCount(3);
+        System.out.println("Vector created in memory: " + intVector);
+    }
+
+**Varchar Vector**: Create an value vector of string like this [one, two, three]
+
+.. code-block:: Java
+
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.vector.VarCharVector;
+
+    try(BufferAllocator rootAllocator = new RootAllocator();
+        VarCharVector varCharVector = new VarCharVector("variable-size-primitive-layout", rootAllocator)){
+        varCharVector.allocateNew(3);
+        varCharVector.set(0, "one".getBytes());
+        varCharVector.set(1, "two".getBytes());
+        varCharVector.set(2, "three".getBytes());
+        varCharVector.setValueCount(3);
+        System.out.println("Vector created in memory: " + varCharVector);
+    }
+
+Create A Field
+**************
+
+Fields are used to denote the particular columns of tabular data.
+
+**Field**: Create a column "document" of string type with metadata.
+
+.. code-block:: Java
+
+    import org.apache.arrow.vector.types.pojo.ArrowType;
+    import org.apache.arrow.vector.types.pojo.Field;
+    import org.apache.arrow.vector.types.pojo.FieldType;
+
+    Map<String, String> metadata = new HashMap<>();
+    metadata.put("A", "Id card");
+    metadata.put("B", "Passport");
+    metadata.put("C", "Visa");
+    Field document = new Field("document", new FieldType(true, new ArrowType.Utf8(), /*dictionary*/ null, metadata), /*children*/ null);
+
+Create A Schema
+***************
+
+Schema holds a sequence of fields together with some optional metadata.
+
+**Schema**: Create a schema describing datasets with two columns:
+a int32 column "A" and a utf8-encoded string column "B"
+
+.. code-block:: Java
+
+    import org.apache.arrow.vector.types.pojo.ArrowType;
+    import org.apache.arrow.vector.types.pojo.Field;
+    import org.apache.arrow.vector.types.pojo.FieldType;
+    import org.apache.arrow.vector.types.pojo.Schema;
+    import static java.util.Arrays.asList;
+
+    Map<String, String> metadata = new HashMap<>();
+    metadata.put("K1", "V1");
+    metadata.put("K2", "V2");
+    Field a = new Field("A", FieldType.nullable(new ArrowType.Int(32, true)), null);
+    Field b = new Field("B", FieldType.nullable(new ArrowType.Utf8()), null);
+    Schema schema = new Schema(asList(a, b), metadata);
+
+Create A VectorSchemaRoot
+***************************
+
+VectorSchemaRoot is somewhat analogous to tables and record batches in the other
+Arrow implementations.
```

Comment:
This doesn't explain what it is in principle. Look to the Python docs for inspiration on how we can talk about things. For instance the Python docs say

> Multiple arrays can be combined in tables to form the columns in tabular data when attached to a column name

So we can say "VectorSchemaRoot combines ValueVectors with a Schema to represent tabular data" or something like that. We can still say it's analogous to tables, but we can't use that as the primary explanation.

Reply:
Added